qiskit/pulse/utils.py

```diff
@@ -0,0 +1,106 @@
+from .interfaces import ScheduleComponent
+from .schedule import Schedule
+
+
+def align_measures(schedule: ScheduleComponent, cmd_def: List[Command],
+                   pulse_library: List[PulseLibraryItem]) -> Schedule:
+    """Return a new schedule where measurements occur at the same physical time, with the remaining
+    schedules appropriately offset. Minimum measurement wait time (to allow for calibration pulses)
+    is enforced.
+
+    Args:
+        schedule: Schedule to be aligned
+        cmd_def: Command definition list
+        pulse_library: A list of pulse definitions
+    """
+    new_schedule = Schedule()
+
+    pulse_library = {p.name: p.samples for p in pulse_library}
+
+    # Need time to allow for calibration pulses to be played for result classification
+    max_calibration_duration = 0
+    for cmd in cmd_def:
+        if cmd.name == 'u2':
+            duration = sum([len(pulse_library[pulse.name]) for pulse in cmd.sequence
+                            if pulse.name in pulse_library])
+            max_calibration_duration = max(duration, max_calibration_duration)
+
+    last_acquire = max([time for time, inst in schedule.instructions
+                        if isinstance(inst, AcquireInstruction)])
+    # Schedule the acquires to be either at the end of the needed calibration time, or when the
+    # last acquire is scheduled, whichever comes later
+    acquire_scheduled_time = max(max_calibration_duration, last_acquire)
+
+    # Shift instruction times according to the new scheduled time for the acquires
+    extra_delays = {}
+    for time, inst in schedule.instructions:
+        if isinstance(inst, AcquireInstruction):
+            new_schedule |= inst << acquire_scheduled_time
+            indices = [a.index for a in inst.acquires]
+            extra_delays.update({i: acquire_scheduled_time - time for i in indices})
+        else:
+            delay = max([extra_delays.get(c.index, 0) for c in inst.timeslots.channels])
+            # TODO: increase the delay on the non max channels?
+            new_schedule |= inst << time + delay
+
+    return new_schedule
+
+
+def replace_implicit_acquires(schedule: ScheduleComponent, meas_map: List[List[int]]) -> Schedule:
+    """Return a new schedule with implicit acquires from the measurement mapping replaced by
+    explicit ones.
+
+    Warning:
+        Since new acquires are being added, Memory Slots will be set to match the qubit index. This
+        may overwrite your specification.
+
+    Args:
+        schedule: Schedule to be aligned
+        meas_map: List of lists of qubits that are measured together
+    """
+    new_schedule = Schedule(name=schedule.name)
+
+    for time, inst in schedule.instructions:
+        if isinstance(inst, AcquireInstruction):
+            # Get the label of all qubits that are measured with the qubit(s) in this instruction
+            meas_group = [g for g in meas_map if inst.acquires[0].index in g][0]
```

Comment:
Does this only check the first acquire channel of each `AcquireInstruction`? I think we roughly want meas_group to roughly be the union of all `meas_map` elements that share at least one element with `[acq.index for acq in inst.acquires`.